runtime/lua/vim/lsp/protocol.lua

```diff
@@ -961,6 +962,27 @@ function protocol.resolve_capabilities(server_capabilities)
     error("The server sent invalid implementationProvider")
   end
 
+  local workspace = server_capabilities.workspace
+  if workspace == nil or workspace.workspaceFolders == nil then
+    -- Defaults if omitted.
+    workspace_properties = {
+      workspace_folder_properties =  {
+        supported = false;
+        changeNotifications=false;
+      }
+    }
+  elseif type(workspace.workspaceFolders) == 'table' then
```

Comment:
I think you can tbl_deepextend for this or something like that maybe?

Reply:
Sorry, where? I just copied the format of the other sections.